core/src/main/java/com/linecorp/armeria/internal/common/AbstractContextAwareFuture.java

```diff
@@ -88,18 +83,12 @@ protected final <I, U, V> BiFunction<I, U, V> makeContextAwareLoggingException(B
     private void makeContextAwareLoggingException0(Runnable action) {
         try (SafeCloseable ignored = context.push()) {
             action.run();
-        } catch (Throwable th) {
-            logger.warn("An error occurred while pushing a context", th);
-            throw th;
```

Comment:
Could you log if an exception is raised while pushing the context?
```java
final SafeCloseable handle;
try {
    handle = context.push();
} catch (Throwable th) {
    logger.warn("An error occurred while pushing a context", th);
    throw th;
}

try {
    action.run();
} finally {
    handle.close();
}
```

Reply:
Sure thing. I thought it might be unnecessary since we already log here
https://github.com/line/armeria/blob/a8a9ca2f3baf3f001b73a2decbcc9c38e3da2e23/core/src/main/java/com/linecorp/armeria/internal/common/RequestContextUtil.java#L107-L109

Reply:
I think it's different logs.
The log in the `newIllegalContextPushingException` method is for a context leak, meanwhile, this log is for other exceptions such as those raised while calling a hook. 😉